scripts/publish.js

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable camelcase */
+import inquirer from "inquirer";
+import { invalid, section, valid } from "../utils.js";
+import ora from "ora";
+import { apiClient } from "./utils/apiClient.js";
+
+const POLL_INTERVAL = 2000;
+
+const IN_PROGRESS_STATUSES = ["PENDING", "STARTED"];
+
+const waitForTaskToComplete = async (taskId) => {
+  const waitingSpinner = ora(
+    "Waiting for publish to complete. This may take a few minutes.\n\n Publishing will continue if you exit here (CTRL+C)."
+  ).start();
+
+  let latestProcessingResultBody;
+
+  // Poll indefinitely until the task is resolved.
+  while (true) {
+    await new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, POLL_INTERVAL);
+    });
+
+    const processingResult = await apiClient.get({
+      path: "/v2/status-of-published-module/" + taskId
+    });
+
+    if (!processingResult.ok) {
+      waitingSpinner.stop();
+
+      // TODO - ask the BE to supply a 400 when there's a regular failure?
+      if (processingResult.status === 500) {
+        const result = await processingResult.text();
+        invalid(`
+Crowdbotics CLI has encountered an error while waiting for the publish command to complete. Publishing may still finish successfully, however you may contact support with reference id: '${taskId}'
+Error details: ${result}
+`);
+      }
+
+      invalid(
+        `Crowdbotics CLI has encountered an error while waiting for the publish command to complete. Publishing may still finish successfully, however you may contact support with reference id: '${taskId}'`
+      );
+      break;
+    }
+
+    latestProcessingResultBody = await processingResult.json();
+
+    // There were some instances while testing where status was coming back as undefined. This may not be an issue anymore.
+    if (!latestProcessingResultBody.status) {
+      waitingSpinner.stop();
+      invalid(
+        `Crowdbotics CLI has encountered an error while waiting for the publish command to complete. Publishing may still finish successfully, however you may contact support with reference id: '${taskId}'`
```

Comment:
This text has been duplicated a few times. Would like to see if we can clean this up once we address the 500 vs 200 response for failed tasks.